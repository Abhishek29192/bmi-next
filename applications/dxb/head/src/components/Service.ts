import { graphql } from "gatsby";
import { RoofProLevel } from "@bmi/company-details";

export type ServiceType = RooferTypesEnum | BranchTypesEnum | MerchantTypesEnum;

export enum RooferTypesEnum {
  PITCHED_ROOF = "Pitched roof",
  FLAT_ROOF = "Flat roof",
  PITCHED_ROOF_BITUMEN_ROOFS = "Pitched roof - Bitumen roofs",
  PITCHED_ROOF_TILE_ROOFS = "Pitched roof - Tile roofs",
  PITCHED_ROOF_ROOF_COATERS = "Pitched roof - Roof coaters"
}

export const rooferTypes = [
  RooferTypesEnum.PITCHED_ROOF,
  RooferTypesEnum.FLAT_ROOF,
  RooferTypesEnum.PITCHED_ROOF_BITUMEN_ROOFS,
  RooferTypesEnum.PITCHED_ROOF_TILE_ROOFS,
  RooferTypesEnum.PITCHED_ROOF_ROOF_COATERS,
  RooferTypesEnum.PITCHED_ROOF_ROOF_COATERS
];

export enum BranchTypesEnum {
  HEADQUARTERS = "Headquarters",
  COUNTRY_OFFICES = "Country offices",
  MANUFACTURING_AND_EXPORT = "Manufacturing and export",
  TECHNICAL_CENTRE_BMI_ACADEMY = "Technical centre & BMI academy"
}

export const branchTypes = [
  BranchTypesEnum.HEADQUARTERS,
  BranchTypesEnum.COUNTRY_OFFICES,
  BranchTypesEnum.MANUFACTURING_AND_EXPORT,
  BranchTypesEnum.TECHNICAL_CENTRE_BMI_ACADEMY
];

export enum MerchantTypesEnum {
  BMI_REDLAND_PITCHED_ROOF_SYSTEMS = "BMI Redland Pitched roof systems",
  BMI_ICOPAL_FLAT_ROOF_SYSTEMS = "BMI Icopal Flat roof Systems",
  DEPOTS_DE_PROXIMITE_SIPLAST = "Dépôts de proximité Siplast",
  DISTRIBUTEURS_SIPLAST = "Distributeurs Siplast",
  DISTRIBUTEURS_MONIER = "Distributeurs Monier",
  FLACHDACHSYSTEME = "Flachdachsysteme",
  STEILDACHSYSTEME = "Steildachsysteme",
  BAUWERKSABDICHTUNG = "Bauwerksabdichtung",
  VERKEHRSBAUWERKE = "Verkehrsbauwerke",
  VERARBEITER = "Verarbeiter",
  TOPDACH_PARTNER = "TopDach Partner",
  DACHDECKER_7 = "7° Dachdecker",
  HANDLER = "Händler"
}

export const merchantTypes = [
  MerchantTypesEnum.BMI_REDLAND_PITCHED_ROOF_SYSTEMS,
  MerchantTypesEnum.BMI_ICOPAL_FLAT_ROOF_SYSTEMS,
  MerchantTypesEnum.DEPOTS_DE_PROXIMITE_SIPLAST,
  MerchantTypesEnum.DISTRIBUTEURS_SIPLAST,
  MerchantTypesEnum.DISTRIBUTEURS_MONIER,
  MerchantTypesEnum.DISTRIBUTEURS_MONIER,
  MerchantTypesEnum.FLACHDACHSYSTEME,
  MerchantTypesEnum.STEILDACHSYSTEME,
  MerchantTypesEnum.BAUWERKSABDICHTUNG,
  MerchantTypesEnum.VERKEHRSBAUWERKE,
  MerchantTypesEnum.VERARBEITER,
  MerchantTypesEnum.TOPDACH_PARTNER,
  MerchantTypesEnum.DACHDECKER_7,
  MerchantTypesEnum.HANDLER
];

export enum EntryTypeEnum {
  ROOFER_TYPE = "Roofer",
  BRANCH_TYPE = "Branch",
  MERCHANT_TYPE = "Merchant"
}

export enum ServiceTypesPrefixesEnum {
  Roofer = "findARoofer",
  Branch = "findABranch",
  Merchant = "findAMerchant"
}

export enum ServiceTypesEnum {
  type = "Roofer",
  branchType = "Branch",
  merchantType = "Merchant"
}

export const serviceTypesByEntity = (entityType: EntryTypeEnum) => {
  switch (entityType) {
    case EntryTypeEnum.ROOFER_TYPE:
      return rooferTypes;
    case EntryTypeEnum.BRANCH_TYPE:
      return branchTypes;
    case EntryTypeEnum.MERCHANT_TYPE:
      return merchantTypes;
    default:
      return [];
  }
};

export type Data = {
  __typename: "ContentfulService";
  id: string;
  entryType: EntryTypeEnum;
  name: string;
  location: {
    lat: number;
    lon: number;
  };
  address: string;
  phone: string | null;
  email: string | null;
  website: string | null;
  fax: string | null;
  type: RooferTypesEnum[] | null;
  branchType: BranchTypesEnum[] | null;
  merchantType: MerchantTypesEnum[] | null;
  certification: RoofProLevel | null;
  summary: string | null;
};

export const query = graphql`
  fragment ServiceFragment on ContentfulService {
    __typename
    id
    entryType
    name
    location {
      lat
      lon
    }
    address
    phone
    email
    website
    fax
    type
    branchType
    merchantType
    certification
    summary
  }
`;
