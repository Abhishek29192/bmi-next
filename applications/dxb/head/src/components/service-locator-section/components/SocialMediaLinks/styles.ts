import Link from "@mui/material/Link";
import { styled } from "@mui/material/styles";
import Icon from "../../../../components/Icon";

export const Anchor = styled("a")({
  alignItems: "center",
  cursor: "pointer",
  display: "flex",
  justifyContent: "center"
});

export const Root = styled("ul")({
  alignItems: "start",
  display: "flex",
  listStyle: "none",
  margin: 0,
  padding: 0
});

export const Item = styled("li")({
  marginRight: "11px",
  "&:last-child": {
    marginRight: 0
  }
});

export const StyledIcon = styled(Icon)(({ theme }) => ({
  fill: theme.colours.cyan500,
  height: "22px",
  width: "22px"
}));

export const StyledLink = styled(Link)({
  alignItems: "center",
  cursor: "pointer",
  display: "flex",
  justifyContent: "center"
});
