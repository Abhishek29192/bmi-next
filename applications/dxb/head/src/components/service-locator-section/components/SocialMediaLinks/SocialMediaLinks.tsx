import { LinkProps } from "@mui/material";
import React from "react";
import withGTM from "../../../../utils/google-tag-manager";
import { Service } from "../../index";
import { getSocialMediaGtm } from "../../utils/getSocialMediaGtm";
import { Item, Root, StyledIcon, StyledLink } from "./styles";

export interface Props {
  "data-testid"?: string;
  service: Service;
}

export const WithGtmLink = withGTM<LinkProps>(StyledLink);

export const SocialMediaLinks = ({
  "data-testid": dataTestId,
  service
}: Props) => {
  const { facebook, instagram, linkedIn, twitter } = service;

  return (
    <Root data-testid={dataTestId ?? "social-media-links"}>
      {facebook && (
        <Item>
          <WithGtmLink
            aria-label="facebook"
            href={`https://www.facebook.com/${facebook}`}
            gtm={getSocialMediaGtm({ channel: "facebook", service })}
          >
            <StyledIcon name="Facebook" />
          </WithGtmLink>
        </Item>
      )}
      {twitter && (
        <Item>
          <WithGtmLink
            aria-label="twitter"
            href={`https://www.twitter.com/${twitter}`}
            gtm={getSocialMediaGtm({ channel: "twitter", service })}
          >
            <StyledIcon name="Twitter" />
          </WithGtmLink>
        </Item>
      )}
      {instagram && (
        <Item>
          <WithGtmLink
            aria-label="instagram"
            href={`https://www.instagram.com/${instagram}`}
            gtm={getSocialMediaGtm({ channel: "instagram", service })}
          >
            <StyledIcon name="Instagram" />
          </WithGtmLink>
        </Item>
      )}
      {linkedIn && (
        <Item>
          <WithGtmLink
            aria-label="linkedIn"
            href={`https://www.linkedin.com/${linkedIn}`}
            gtm={getSocialMediaGtm({ channel: "linkedIn", service })}
          >
            <StyledIcon name="LinkedIn" />
          </WithGtmLink>
        </Item>
      )}
    </Root>
  );
};
