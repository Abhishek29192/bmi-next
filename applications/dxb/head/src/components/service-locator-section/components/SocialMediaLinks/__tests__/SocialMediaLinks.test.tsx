import { screen } from "@testing-library/react";
import React from "react";
import createRoofer from "../../../../../__tests__/helpers/ServiceHelper";
import { renderWithProviders } from "../../../../../__tests__/renderWithProviders";
import { Data as Service } from "../../../../Service";
import { Props, SocialMediaLinks } from "../SocialMediaLinks";

const service = createRoofer();

const props: Props = {
  service
};

const channels: Partial<Service> = {
  facebook: "boop",
  instagram: "beep",
  linkedIn: "bang",
  twitter: "boom"
};

describe("SocialMediaLinks component", () => {
  it("renders correctly", () => {
    const { container } = renderWithProviders(<SocialMediaLinks {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('has the default "data-testid" attribute', () => {
    renderWithProviders(<SocialMediaLinks {...props} />);
    expect(screen.getByTestId("social-media-links")).toBeInTheDocument();
  });

  Object.keys(channels).forEach((channel) => {
    describe(`When a ${channel} link is provided`, () => {
      const updatedProps = {
        ...props,
        service: { ...props.service, ...channels }
      };

      it(`displays a ${channel} link`, () => {
        renderWithProviders(<SocialMediaLinks {...updatedProps} />);

        expect(
          screen.getByLabelText(channel, { exact: false })
        ).toBeInTheDocument();
      });

      it("uses the correct link url", () => {
        renderWithProviders(<SocialMediaLinks {...updatedProps} />);
        expect(screen.getByLabelText(channel)).toHaveAttribute(
          "href",
          `https://www.${channel.toLowerCase()}.com/${
            channels[channel as string]
          }`
        );
      });
    });

    describe(`When a ${channel} link is NOT provided`, () => {
      const updatedChannels = Object.keys(channels).reduce(
        (acc, curr) =>
          curr === channel ? acc : { ...acc, [curr]: channels[curr as string] },
        {}
      );

      const updatedProps = {
        ...props,
        service: { ...props.service, ...updatedChannels }
      };
      it(`does NOT display a ${channel} link`, () => {
        renderWithProviders(<SocialMediaLinks {...updatedProps} />);

        expect(
          screen.queryByTestId(`${channel}Icon`, { exact: false })
        ).not.toBeInTheDocument();
      });
    });
  });
});
