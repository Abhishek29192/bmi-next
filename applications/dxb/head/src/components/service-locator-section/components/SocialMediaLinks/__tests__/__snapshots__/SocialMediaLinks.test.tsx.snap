// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMediaLinks component renders correctly 1`] = `
<div>
  <ul
    class="css-1nn191a"
    data-testid="social-media-links"
  />
</div>
`;
